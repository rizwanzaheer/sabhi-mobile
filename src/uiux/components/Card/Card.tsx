import * as React from 'react'
import { TouchableHighlight, ViewStyle } from 'react-native'
import Container from '../Container/Container'
import * as Kancha from '../../types'
import { withTheme } from '../../theming'
import Device from '../../services/device'

interface CardProps {
  onPress?: () => void
  shadow?: number
  br?: number
  testID?: string
  background?: Kancha.BrandPropOptions
  marginBottom?: number
  theme: any
}

const Card: React.FC<CardProps> = ({
  children,
  onPress,
  shadow,
  br,
  background,
  theme,
  testID,
  marginBottom,
}) => {
  const style: ViewStyle = {
    borderRadius: br || 5,
    ...(Device.isAndroid ? { marginRight: 8 } : {}),
    backgroundColor: background && theme.colors[background].background,
    marginBottom: marginBottom || theme.spacing.default,
    ...(shadow
      ? {
          shadowColor: '#000000',
          shadowRadius: shadow * 4,
          shadowOpacity: shadow / 15,
          elevation: 5,
        }
      : {}),
  }
  return (
    <TouchableHighlight
      onPress={onPress}
      disabled={!onPress}
      testID={testID}
      style={style}
      underlayColor={'transparent'}
    >
      <Container padding={true} marginBottom={true}>
        {children}
      </Container>
    </TouchableHighlight>
  )
}

export default withTheme(Card)
