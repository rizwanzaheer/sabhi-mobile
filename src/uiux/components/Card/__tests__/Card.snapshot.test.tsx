import * as React from 'react'
import { render, fireEvent } from 'react-native-testing-library'
import Card from '../Card'
import Device from '../../../services/device'

describe('Component(snapshot): Card', () => {
  it('should render with default props - Android', () => {
    Device.isAndroid = true
    const tree = render(<Card />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with default props', () => {
    const tree = render(<Card />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should render with props', () => {
    const onPress = jest.fn()
    const tree = render(<Card onPress={onPress} shadow={0.3} br={10} marginBottom={10} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('should fire tap event on card when onPress is passed', () => {
    const onPress = jest.fn()
    const { getByTestId } = render(<Card onPress={onPress} testID={'CARD_BTN'} />)

    fireEvent.press(getByTestId('CARD_BTN'))
    expect(onPress).toHaveBeenCalled()
  })
})
