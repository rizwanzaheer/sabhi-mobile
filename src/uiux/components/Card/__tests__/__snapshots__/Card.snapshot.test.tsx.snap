// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component(snapshot): Card should render with default props - Android 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
      "borderRadius": 5,
      "marginBottom": 15,
      "marginRight": 8,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": 15,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 15,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  />
</View>
`;

exports[`Component(snapshot): Card should render with default props 1`] = `
<View
  accessible={true}
  focusable={false}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
      "borderRadius": 5,
      "marginBottom": 15,
      "marginRight": 8,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": 15,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 15,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  />
</View>
`;

exports[`Component(snapshot): Card should render with props 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": undefined,
      "borderRadius": 10,
      "elevation": 5,
      "marginBottom": 10,
      "marginRight": 8,
      "shadowColor": "#000000",
      "shadowOpacity": 0.02,
      "shadowRadius": 1.2,
    }
  }
>
  <View
    pointerEvents="auto"
    style={
      Object {
        "alignItems": undefined,
        "backgroundColor": undefined,
        "borderRadius": undefined,
        "flex": undefined,
        "flexDirection": undefined,
        "height": undefined,
        "justifyContent": undefined,
        "margin": undefined,
        "marginBottom": 15,
        "marginLeft": undefined,
        "marginRight": undefined,
        "marginTop": undefined,
        "padding": 15,
        "paddingBottom": undefined,
        "paddingHorizontal": undefined,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "width": undefined,
      }
    }
  />
</View>
`;
